MapboxNavigation/NavigationMapView.swift

```diff
@@ -1335,3 +1367,122 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         enableFrameByFrameCourseViewTracking(for: 3)
     }
 }
+
+// MARK: - Building Extrusion Highlights
+
+private struct BuildingHighlightAttributes: Hashable {
+    static let kDefaultIdentifier: Int64 = -1
+    
+    var coordinate: CLLocationCoordinate2D = kCLLocationCoordinate2DInvalid
+    var highlightColor: UIColor = .defaultBuildingHighlightColor
+    var identifier: Int64 = kDefaultIdentifier
+    
+    func hash(into hasher: inout Hasher) {
+        hasher.combine(identifier)
+    }
+    
+    static func ==(lhs: BuildingHighlightAttributes, rhs: BuildingHighlightAttributes) -> Bool {
+        return lhs.identifier == rhs.identifier
+    }
+}
+
+extension NavigationMapView {
+       
+    /**
+     Receives coordinates for searching the map for buildings. If buildings are found, they will be highlighted in 2d or 3d depending on the `in3D` value.
+     `in3D` defaults to true
+     `buildingHighlightingEnabled` must be set to true before this function will work.
+     */
+    public func highlightBuildings(at coordinates: [CLLocationCoordinate2D], in3D extrudesBuildings: Bool = true) {
```

Comment:
Should this method return any sort of indication as to whether it found a building that can be extruded? That would enable the caller to stop calling this method on an interval.

Reply:
Not sure if there is such need at this point. During implementation of some kind of polling mechanism to check whether there are any buildings which must be rendered it can be revised.